### PR TITLE
xmake: add page

### PR DESCRIPTION
**Pull request by theKnightsOfRohan**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: pages/common/xmake.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# xmake
+
+> A cross-platform C & C++ build utility based on lua.
+> More information: <https://xmake.io/#/getting_started>.
+
+- Create an xmake C project, consisting of a hello world and xmake.lua:
+
+`xmake create --language c -P {{any-project-name}}`
```
Comment (spageektti) [inline pages/common/xmake.md:8]:
```suggestion
`xmake create --language c -P {{any-project-name}}`
```

```suggestion
`xmake create --language c -P {{project-name}}`
```

File: pages/common/xmake.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- Build and run an xmake project:
+
+`xmake build run`
+
+- Run a compiled xmake target directly:
+
+`xmake run {{targetname}}`
```
Comment (spageektti) [inline pages/common/xmake.md:16]:
```suggestion
`xmake run {{target_name}}`
```

```suggestion
`xmake run {{target-name}}`
```

```suggestion
`xmake run {{target}}`
```

File: pages/common/xmake.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+- Configure an project's build targets:
+
+`xmake config --plat={{macosx|linux|iphoneos ..}} --arch={{x86_64|i386|arm64 ..}} --mode={{debug|release}}`
```
Comment (spageektti) [inline pages/common/xmake.md:20]:
```suggestion
`xmake config --plat={{macosx|linux|iphoneos|...}} --arch={{x86_64|i386|arm64|...}} --mode={{debug|release}}`
```

File: pages/common/xmake.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+- Install the compiled target to an output directory:
+
+`xmake install -o {{installdir}}`
```
Comment (spageektti) [inline pages/common/xmake.md:24]:
```suggestion
`xmake install -o {{path/to/directory}}`
```

File: pages/common/xmake.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# xmake
+
+> A cross-platform C & C++ build utility based on lua.
```
Comment (spageektti) [inline pages/common/xmake.md:3]:
```suggestion
> A cross-platform C & C++ build utility based on Lua.
```

**Review by spageektti — COMMENTED** [review]
Hi. Thanks for your contribution. I left some suggestions.

File: pages/common/xmake.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+- Build and run an xmake project:
+
+`xmake build run`
+
+- Run a compiled xmake target directly:
+
+`xmake run {{target_name}}`
+
+- Configure an project's build targets:
```
Comment (gutjuri) [inline pages/common/xmake.md:18]:
```suggestion
- Configure a project's build targets:
```

**Review by gutjuri — APPROVED** [review]
Thanks! Looks good to me after fixing a type ;)

**Review by spageektti — APPROVED** [review]

File: pages/common/xmake.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# xmake
+
+> A cross-platform C & C++ build utility based on Lua.
+> More information: <https://xmake.io/#/getting_started>.
+
+- Create an xmake C project, consisting of a hello world and xmake.lua:
```
Comment (acuteenvy) [inline pages/common/xmake.md:6]: Xmake is written with a capital letter in the project's README

```suggestion
- Create an Xmake C project, consisting of a hello world and `xmake.lua`:
```

File: pages/common/xmake.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+`xmake create --language c -P {{project-name}}`
```
Comment (acuteenvy) [inline pages/common/xmake.md:8]: https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md#naming
> Use `snake_case` for multi-word placeholders.
```suggestion
`xmake create --language c -P {{project_name}}`
```

File: pages/common/xmake.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+- Build and run an xmake project:
```
Comment (acuteenvy) [inline pages/common/xmake.md:10]:
```suggestion
- Build and run an Xmake project:
```

File: pages/common/xmake.md [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+`xmake build run`
+
+- Run a compiled xmake target directly:
```
Comment (acuteenvy) [inline pages/common/xmake.md:14]:
```suggestion
- Run a compiled Xmake target directly:
```

File: pages/common/xmake.md [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+
+`xmake run {{target_name}}`
+
+- Configure a project's build targets:
+
+`xmake config --plat={{macosx|linux|iphoneos|...}} --arch={{x86_64|i386|arm64|...}} --mode={{debug|release}}`
+
+- Install the compiled target to an output directory:
```
Comment (acuteenvy) [inline pages/common/xmake.md:22]:
```suggestion
- Install the compiled target to a directory:
```

**Pull request merged**